### PR TITLE
Adding converter samples

### PR DESCRIPTION
**Pull request by Hackavist**
### Description of Change ###
Added the Stated converters in #430 

### Bugs Fixed ###

Added missing Converters to sample app.
Code House Keeping
### API Changes ###
No api changes

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard

- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

File: samples/XCT.Sample/ViewModels/Converters/DoubleToIntConverterViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	public class DoubleToIntConverterViewModel : BaseViewModel
+	{
+		double index;
+		public double Input
+		{
+			get => index;
+			set
+			{
+				index = value;
+				OnPropertyChanged(nameof(Input));
```
Comment (pictos) [inline samples/XCT.Sample/ViewModels/Converters/DoubleToIntConverterViewModel.cs:13]: We don't need the `nameof` here, you can change this and others?
```suggestion
				OnPropertyChanged();
```

**Review by pictos — CHANGES_REQUESTED** [review]
Thanks for this! I have just one request

**Comment by Hackavist** [comment]
@pictos thank you for taking the time to review my pr , i have committed the changes that you requested, if there is any way we can further optimize the code please tell me.

File: samples/XCT.Sample/ViewModels/Converters/ConvertersGalleryViewModel.cs [hunk 2]
```diff
@@ -10,15 +10,54 @@ public class ConvertersGalleryViewModel : BaseGalleryViewModel
...
+				"The DoubleToIntConverter is a converter that allows users to convert an incoming double value to an int."),
+
+			new SectionModel(
+				typeof(IndexToArrayItemConverterPage),
+				nameof(IndexToArrayItemConverter),
+				"The IndexToArrayItemConverter is a converter that allows users to convert a int value binding to an item in an array."),
+			new SectionModel(
+				typeof(IntToBoolConverterPage),
+				nameof(IntToBoolConverter),
+				"The IntToBoolConverter is a converter that allows users to convert an incoming int value to a bool."),
```
Comment (AndreiMisiukevich) [inline samples/XCT.Sample/ViewModels/Converters/ConvertersGalleryViewModel.cs:50]: Maybe let's go ahead with "A converter that..." as a start of the description?

File: samples/XCT.Sample/ViewModels/Converters/IsNullOrEmptyConverterViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+			{
+				selectedItem = value;
+				OnPropertyChanged();
+			}
+		}
+
+		public ICommand ClearSelectionCommand => new Command(() =>
+		{
+			SelectedItem = null;
+		});
```
Comment (AndreiMisiukevich) [inline samples/XCT.Sample/ViewModels/Converters/IsNullOrEmptyConverterViewModel.cs:34]:
```suggestion
		ICommand clearSelectionCommand;
		public ICommand ClearSelectionCommand => clearSelectionCommand ??= new Command(() =>
		{
			SelectedItem = null;
		});
```

File: samples/XCT.Sample/ViewModels/Converters/DoubleToIntConverterViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	{
+		double index;
+		public double Input
+		{
+			get => index;
+			set
+			{
+				index = value;
+				OnPropertyChanged();
+			}
```
Comment (AndreiMisiukevich) [inline samples/XCT.Sample/ViewModels/Converters/DoubleToIntConverterViewModel.cs:14]: Please use ```SetProperty``` in all view models :)

```set => SetProperty(ref index, value);```

**Review by AndreiMisiukevich — CHANGES_REQUESTED** [review]
Could you please address all comments and update the code?)
Thanks

File: samples/XCT.Sample/ViewModels/Converters/IsNullOrEmptyConverterViewModel.cs (hunk 3 above)
Comment (Cfun1) [reply samples/XCT.Sample/ViewModels/Converters/IsNullOrEmptyConverterViewModel.cs:34]: I knew "??" but not "??=" good to know about 👍🏻

**Comment by Hackavist** [comment]
@AndreiMisiukevich changes met

**Review by pictos — APPROVED** [review]
For me, this looks good. Thanks @Hackavist

**Comment by Hackavist** [comment]
@pictos  Thank you for taking the time to review it, it was my pleasure working on it :D

File: samples/XCT.Sample/ViewModels/Converters/ConvertersGalleryViewModel.cs [hunk 5]
```diff
@@ -45,6 +77,14 @@ public class ConvertersGalleryViewModel : BaseGalleryViewModel
...
 				nameof(ListIsNullOrEmptyConverter),
 				"A converter that allows you to check if collection is null or empty"),
+			new SectionModel(
+				typeof(ListIsNotNullOrEmptyConverterPage),
+				nameof(ListIsNotNullOrEmptyConverter),
+				"A converter that allows you to check if collection is not null or empty"),
+			new SectionModel(
+				typeof(ListToStringConverterPage),
+				nameof(ListToStringConverter),
+				"The ListToStringConverter is a converter that allows users to convert an incoming binding that implements IEnumerable to a single string value. The Separator property is used to join the items in the IEnumerable."),
```
Comment (AndreiMisiukevich) [inline samples/XCT.Sample/ViewModels/Converters/ConvertersGalleryViewModel.cs:87]:
```suggestion
				"A converter that allows users to convert an incoming binding that implements IEnumerable to a single string value. The Separator property is used to join the items in the IEnumerable."),
```

File: samples/XCT.Sample/ViewModels/Converters/IsNullOrEmptyConverterViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+			get => selectedItem;
+			set => SetProperty(ref selectedItem, value);
+		}
+
+		ICommand clearSelectionCommand;
+
+		public ICommand ClearSelectionCommand => clearSelectionCommand ??= new Command(() =>
+		{
+			SelectedItem = null;
+		});
```
Comment (AndreiMisiukevich) [inline samples/XCT.Sample/ViewModels/Converters/IsNullOrEmptyConverterViewModel.cs:33]:
```suggestion
		public ICommand ClearSelectionCommand => clearSelectionCommand ??= new Command(() => SelectedItem = null);
```

**Review by AndreiMisiukevich — APPROVED** [review]

**Review by pictos — APPROVED** [review]

**Pull request merged**